rcm.html

```diff
@@ -1,118 +1,90 @@
-  arpeggios["p-a-m-i descending"]                 = "p-i-m-a ascending";
-  arpeggios["p-i-m-i pattern"]                    = "p-m-i-m pattern ";
-  arpeggios["p-m-i-m pattern"]                    = "p-i-m-i pattern";
-  arpeggios["ascending and descending sextuplet"] = "descending and ascending sextuplet";
-  arpeggios["descending and ascending sextuplet"] = "ascending and descending sextuplet";
-  arpeggios["p-a dyad-m-i triplet pattern"]       = "p-i dyad-m-a triplet pattern";
-  arpeggios["p-i dyad-m-a triplet pattern"]       = "p-a dyad-m-i triplet pattern";
-  arpeggios["p-m dyad-i-a triplet pattern"]       = "p-i dyad-a-m triplet pattern";
-  arpeggios["p-i dyad-a-m triplet pattern"]       = "p-m dyad-i-a triplet pattern ";
-
-  var arpeggio1 = get_random_element(Object.keys(arpeggios));
-  var arpeggio2 = arpeggios[arpeggio1];
-
-  var arpeggio_speed = "";
-  if (arpeggio1.indexOf("sextuplet") !== -1)
-  {
-	arpeggio_speed = 52;
-  }
-  else
-  {
-	arpeggio_speed = 76;
-  }
+  const arpeggios = {
+    'p-i-m-a ascending'                  : 'p-a-m-i descending',
+    'p-a-m-i descending'                 : 'p-i-m-a ascending',
+    'p-i-m-i pattern'                    : 'p-m-i-m pattern ',
+    'p-m-i-m pattern'                    : 'p-i-m-i pattern',
+    'ascending and descending sextuplet' : 'descending and ascending sextuplet',
+    'descending and ascending sextuplet' : 'ascending and descending sextuplet',
+    'p-a dyad-m-i triplet pattern'       : 'p-i dyad-m-a triplet pattern',
+    'p-i dyad-m-a triplet pattern'       : 'p-a dyad-m-i triplet pattern',
+    'p-m dyad-i-a triplet pattern'       : 'p-i dyad-a-m triplet pattern',
+    'p-i dyad-a-m triplet pattern'       : 'p-m dyad-i-a triplet pattern ',
+  };
+  const arpeggio1 = getRandomElement(Object.keys(arpeggios));
+  const arpeggio2 = arpeggios[arpeggio1];
+  const arpeggioSpeed = arpeggio1.includes('sextuplet') ? '52' : '76';
 
   // the pieces I am required to memorize
-  var piece = get_random_element(
-    ["Bach Minuet",
-    "Paganini Minuet",
-    "Spanish Romance"]);
+  const piece = getRandomElement([
+    'Bach Minuet',
+    'Paganini Minuet',
+    'Spanish Romance'
+  ]);
 
   // the studies I am required to memorize
-  var study = get_random_element(
-    ["Carcassi Tremolo Study",
-    "Bosch Lamento"]);
+  const study = getRandomElement([
+    'Carcassi Tremolo Study',
+    'Bosch Lamento'
+  ]);
 
   // output!
-  scale_output    = "<b>Scales: </b>" + scale1 + " and " + scale2 + ". " + note_value + ", " + stroke + ", " + fingering;
-  if (tremolando)
-  {
-    scale_output = scale_output + ". Tremolando";	
-  }
-  arpeggio_output = "<b>Arpeggio: </b>" + arpeggio1 + " and " + arpeggio2 + " at " + arpeggio_speed;
-  piece_output    = "<b>Piece: </b>" + piece;
-  study_output    = "<b>Study: </b>" + study;
+  const scaleOutput = `<strong>Scales: </strong> ${scale1} and ${scale2}. ${noteValue}, ${stroke}, ${fingering}${tremolando ? '. Tremolando' : ''}`;
```

Comment:
ES6 also supports template literals with the backtick character, which reduces the "" + "" for concatenating strings